components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/dao/UserAccountAssociationDAO.java

```diff
@@ -144,7 +150,13 @@ public void deleteUserAssociation(String domainName, int tenantId,
                                 (conUserName)) {
                             continue;
                         }
-
+                        if (realmConfig != null &&
```

Comment:
Extract the logic in if condition to a private method named isUserStoreDisabled()